### ISSUE TITLE
Sending null crashed the autoupdate service

### ISSUE THREAD

**Issue by FinnStutzenstein** [OpenSlides/openslides-autoupdate-service#120]
Sending `ids: [NaN]`, results in 500er errors:

```
autoupdate_1        | 2020/10/28 11:32:06 Internal Error: build keysbuilder: build keys: build keys for the first time: load needed keys: get values for keys `[assignment/0/list_of_speakers_id assignment/0/candidate_ids]` from datastore: getOrSet for keys `[assignment/0/list_of_speakers_id assignment/0/candidate_ids]`: fetching key: fetching missing keys: datastore returned status 400 BAD REQUEST: {
autoupdate_1        |   "error": {
autoupdate_1        |     "msg": "The key 'assignment/0/list_of_speakers_id' is no fqid, fqfield or collectionkey", 
autoupdate_1        |     "type": 1, 
autoupdate_1        |     "type_verbose": "INVALID_FORMAT"
autoupdate_1        |   }
autoupdate_1        | }
```

Please handle this case correctly.

**Comment by ostcar** [comment]
Can you send me a full request? I get the following error:

```
$ curl -i -Nk  https://localhost:9012/system/autoupdate -d '[{"ids": [NaN], "collection": "user", "fields": {"username": null}}]'
HTTP/2 400 
content-type: application/octet-stream
content-length: 95
date: Wed, 28 Oct 2020 17:18:47 GMT

{"error": {"type": "JsonError", "msg": "invalid character 'N' looking for beginning of value"}}
```
This is was I would expect.

**Comment by FinnStutzenstein** [comment]
Oh sorry, `NaN` is serialized into `null`. I thought, JSOn had a NaN... So `null` must be handeled (or something else - what happens, if there is another list/object/...?).

```
{"collection":"assignment","ids":[null],"fields":{"title":null,"open_posts":null,"phase":null,"candidate_ids":{"type":"relation-list","collection":"assignment_candidate","fields":{"weight":null,"user_id":{"type":"relation","collection":"user","fields":{"title":null,"username":null,"first_name":null,"last_name":null,"id":null}},"assignment_id":null,"id":null}},"description":null,"default_poll_description":null,"number_poll_candidates":null,"id":null,"list_of_speakers_id":{"type":"relation","collection":"list_of_speakers","fields":{"closed":null,"id":null,"speaker_ids":{"type":"relation-list","collection":"speaker","fields":{"begin_time":null,"end_time":null,"id":null}}}}}}
```

**Comment by ostcar** [comment]
It is a bit difficult to guess what you are telling me. Could you always tell me in an issue what you did, what happend and what you expected?

The problem seams to be https://github.com/golang/go/issues/33835

For go `null == 0`. So this issue is the same, as if you would set the value of `ids` to `[0]`. The datastore thinks, that an fqid with the id being `0` is an invalid key.

**Comment by FinnStutzenstein** [comment]
You should validate the data you get - an id is a positive integer, so it is a client issue and not a server one. The status code of 5xx is not good since it should be a 4xx on a client error.

**Comment by ostcar** [comment]
Should be fixed in #123.
